lib/item_repository.rb

```diff
@@ -0,0 +1,20 @@
+require_relative 'item'
+
+class ItemRepository
+  def all
+    sql = "SELECT * FROM items;"
+    DatabaseConnection.exec_params(sql, []).map do |record|
+      item = Item.new
+      item.id = record['id']
+      item.name = record['name']
+      item.unit_price = record['unit_price']
+      item.quantity = record['quantity']
+      item
+    end
+  end
+
+  def create(item)
+    sql = "INSERT INTO items (name, unit_price, quantity) VALUES ($1, $2, $3);"
+    DatabaseConnection.exec_params(sql, [item.name, item.unit_price, item.quantity])
```

Comment:
This is uses less lines total but potentially less descriptive than using sql_params. I do think that it's good practice not to use a variable for only one line though so I can see why you are doing it this way

Reply:
I second that - seems more readable to me if params are split out on line above - even though this is perfectly readable